src/harness/reference_models/iap/iap.py

```diff
@@ -0,0 +1,654 @@
+
+# Global IAP protection threshold per IAPBW_HZ container (dBm/IAPBW_HZ)
+threshold_per_iapbw = {}
+
+# values from WINNF-TS-0061-V1.1.0 - WG4 SAS Test and Certification Spec-Table 
+# 8.4-2 Protected entity reference for IAP Protection 
+# The values are in (dBm/RBW)
+protection_thresholds = {'QPpa': -80, 'QGwpz': -80, 'QFssCochannel': -129, 
+                         'QFssBlocking': -60, 'QEsc': -109}
+
+# Default 1 dB pre IAP margin for all incumbent types
+pre_iap_headrooms = {'MgPpa': 1, 'MgGwpz': 1, 'MgFssCochannel': 1, 
+                     'MgFssBlocking': 1, 'MgEsc': 1}
+
+
+def iapPointConstraint(protection_point, channels, low_freq, high_freq, 
+                       grant_objects, fss_info, esc_antenna_info, region_type, 
+                       threshold, protection_ent_type):
+  """ Routine to compute aggregate interference(Ap) from authorized grants 
+   
+  Aggregate interference is calculated over all the 5MHz channels.This routine 
+  is applicable for FSS Co-Channel and ESC Sensor protection points.Also for 
+  protection points within PPA and GWPZ protection areas. Post IAP grant EIRP
+  and interference contribution is calculated over each 5MHz channel
+
+  Args:
+    protection_point: List of protection points inside protection area
+    channels: List of tuples containing (low_freq, high_freq) of each 5MHz 
+              segment of the whole frequency range of protected entity
+    low_freq: low frequency of protection constraint (Hz)
+    high_freq: high frequency of protection constraint (Hz)
+    grant_objects: a list of grant requests, each one being a tuple 
+                   containing grant information
+    fss_info: contains information on antenna height and 
+              weights on the tangent and perpendicular components.
+    esc_antenna_info: contains information on ESC antenna height, azimuth, 
+                      gain and pattern gain
+    threshold: protection threshold (dBm/IAPBW_HZ)
+    protection_ent_type: an enum member of class ProtectedEntityType
+  Returns:
+    Updates global container aggregate_interference dictionary in the format of 
+    {(latitude,longitude,low_freq,high_freq,entity_type):aggregate_interference}
+    aggregate_interference value is in mW/RBW.
+    and global container post_iap_managed_grants nested dictionary in the format of
+    {grant_id:{ProtectionConstraint:PostIAPGrantInfo,ProtectionConstraint:PostIAPGrantInfo},
+    where ProtectionConstraint and PostIAPGrantInfo is a namedtuple 
+  """
+
+  for channel in channels:
+    # Set interference quota for protection_point, channel combination
+
+    # Calculate Roll of Attenuation for ESC Sensor Protection entity
+    if protection_ent_type is interf.ProtectedEntityType.ESC_CAT_A or\
+      protection_ent_type is interf.ProtectedEntityType.ESC_CAT_B:
+      if channel[0] >= 3650.e6:
+        center_freq = (channel[0] + channel[1]) / 2
+        interference_quota = interf.dbToLinear(interf.linearToDb(threshold) - 
```

Comment:
Seems wrong to me.
threshold is specified as dBm in the docstring.  but you are doing linearToDb(threshold)
Isnt it the reverse: dbToLinearI() ?? Same for the dbToLinear which should be reversed.

I am surprised this was not seen during testing, since threshold is negative the linear2Db should raise a fatal exception.

Reply:
threshold   is passed in mW. Updated doc string